src/plan_expand_hypertable.c

```diff
@@ -0,0 +1,253 @@
+{
+	if (node == NULL)
+		return false;
+
+	if (IsA(node, FromExpr))
+	{
+		FromExpr   *f = (FromExpr *) node;
+		ListCell   *lc;
+
+		foreach(lc, (List *) f->quals)
+		{
+			Node	   *qual = (Node *) lfirst(lc);
+			RestrictInfo *restrictinfo;
+
+			Relids		relids = pull_varnos(qual);
+
+			if (bms_num_members(relids) != 1 || !bms_is_member(ctx->rel->relid, relids))
+				continue;
+#if PG96
+			restrictinfo = make_restrictinfo((Expr *) qual,
+											 true,
+											 false,
+											 false,
+											 relids,
+											 NULL,
+											 NULL);
+#else
+			restrictinfo = make_restrictinfo((Expr *) qual,
+											 true,
+											 false,
+											 false,
+											 ctx->root->qual_security_level,
+											 relids,
+											 NULL,
+											 NULL);
+#endif
+			ctx->result = lappend(ctx->result, restrictinfo);
+		}
+	}
+
+	return expression_tree_walker(node, collect_quals_walker, ctx);
+}
+
+/* Since baserestrictinfo is not yet set by the planner, we have to derive
+ * it ourselves. It's safe for us to miss some restrict info clauses (this
+ * will just results in more chunks being included) so this does not need
+ * to be as comprehensive as the PG native derivation. This is inspired
+ * by the derivation in `deconstruct_recurse` in PG */
+
+static List *
+get_restrict_info(PlannerInfo *root, RelOptInfo *rel)
+{
+	CollectQualCtx ctx = {
+		.root = root,
+		.rel = rel,
+		.result = NIL,
+	};
+
+	collect_quals_walker((Node *) root->parse->jointree, &ctx);
+
+	return ctx.result;
+}
+
+static List *
+find_children_oids(HypertableRestrictInfo *hri, Hypertable *ht, LOCKMODE lockmode)
+{
+	List	   *result;
+
+	/*
+	 * optimization: using the HRI only makes sense if we are not using all
+	 * the chunks, otherwise using the cached inheritance hierarchy is faster.
+	 */
+	if (!hypertable_restrict_info_has_restrictions(hri))
+		return find_all_inheritors(ht->main_table_relid, lockmode, NULL);;
+
+	/* always include parent again, just as find_all_inheritors does */
+	result = list_make1_oid(ht->main_table_relid);
+
+	/* add chunks */
+	result = list_concat(result,
+						 hypertable_restrict_info_get_chunk_oids(hri,
+																 ht,
+																 lockmode));
+	return result;
+}
+
+bool
+plan_expand_hypertable_valid_hypertable(Hypertable *ht, Query *parse, Index rti, RangeTblEntry *rte)
+{
+	if (ht == NULL ||
+	/* inheritance enabled */
+		rte->inh == false ||
+	/* row locks not necessary */
+		parse->rowMarks != NIL ||
+	/* not update and/or delete */
+		0 != parse->resultRelation)
+		return false;
+
+	return true;
+}
+
+/* Inspired by expand_inherited_rtentry but expands
+ * a hypertable chunks into an append rekationship */
+void
+plan_expand_hypertable_chunks(Hypertable *ht,
+							  PlannerInfo *root,
+							  Oid relation_objectid,
+							  bool inhparent,
+							  RelOptInfo *rel)
+{
+	RangeTblEntry *rte = rt_fetch(rel->relid, root->parse->rtable);
+	List	   *inh_oids;
+	Oid			parent_oid = relation_objectid;
```

Comment:
Why this extra variable? Don't see it set anywhere. Is it a name clarity issue? Then why not just use the name for the function parameter?